src/Analyser/TypeSpecifierContext.php

```diff
@@ -18,17 +19,15 @@ final class TypeSpecifierContext
 	public const CONTEXT_FALSEY = self::CONTEXT_FALSE | self::CONTEXT_FALSEY_BUT_NOT_FALSE;
 	public const CONTEXT_BITMASK = 0b1111;
 
-	/** @var self[] */
-	private static array $registry;
```

Comment:
drop the caching, as every context gets unique because it contains an additional Type